src/models/callercalleemodel.h

```diff
@@ -0,0 +1,67 @@
+/*
+  callercalleemodel.h
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Nate Rogers <nate.rogers@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <QVector>
+#include <QAbstractItemModel>
+
+#include "framedata.h"
+
+class CallerCalleeModel : public QAbstractTableModel
+{
+    Q_OBJECT
+public:
+    explicit CallerCalleeModel(QObject* parent = nullptr);
+    ~CallerCalleeModel();
+
+    enum Columns {
+        Symbol = 0,
+        Binary,
+        Address,
+        Location,
+        SelfCost,
+        InclusiveCost,
+        NUM_COLUMNS
+    };
+
+    enum Roles {
+        SortRole = Qt::UserRole,
+        FilterRole
+    };
+
+    QVariant headerData(int section, Qt::Orientation orientation = Qt::Horizontal,
+                        int role = Qt::DisplayRole) const override;
+    QVariant data(const QModelIndex& index, int role = Qt::DisplayRole) const override;
+
+    int columnCount(const QModelIndex& parent = {}) const override;
+    int rowCount(const QModelIndex &parent = {}) const override;
+
+    void resetData(const QVector<FrameData>& rows);
+private:
+    QVector<FrameData> m_rows;
```

Comment:
I'd say use a FrameData here as "root" which holds the rows in it's `children`. The advantage is that this will allow us to track the total cost in the root object, for use in calculating the fractional cost.